### PR TITLE
Consistent processing when the target file is missing.

### PR DESCRIPTION
**Pull request by MarkGibbons**
Add the ignore_missing option
Set the default for ignore_missing to true
Add tests of processing missing files with and without ignore_missing set to false
This change restores the processing of missing files to the 1.0.0 behavior.  We flipped with with 1.1.0 release and have gone back to the old behavior with 2.0.0

---
This change is [Reviewable](https://reviewable.io/reviews/sous-chefs/line-cookbook/99)

### CODE REVIEW

**Comment by MarkGibbons** [comment]
I'm working on the travis and appveyor updates.

**Comment by MarkGibbons** [comment]
Any idea what I need to do to get circleci to run?

**Comment by damacus** [comment]
Ah I had, build forked projects turned off. Try another push?

File: .kitchen.dokken.yml [hunk 1]
```diff
@@ -26,20 +26,25 @@ platforms:
     pid_one_command: /bin/systemd
 
 suites:
```
Comment (damacus) [inline .kitchen.dokken.yml]: All of the dokken suites can be removed as we use KITCHEN_LOCAL to override certain sections. We'll just manage suites in the main kitchen.yml

File: resources/delete_from_list.rb [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
 property :eol, String, default: Line::OS.unix? ? "\n" : "\r\n"
 property :backup, [true, false], default: false
+property :ignore_missing, [true, false], default: true
 
 resource_name :delete_from_list
 
 action :edit do
-  return if !::File.exist?(new_resource.path) && new_resource.ignore_missing
-  raise "File #{new_resource.path} not found" unless ::File.exist?(new_resource.path)
+  file_exist = ::File.exist?(new_resource.path)
```
Comment (damacus) [inline resources/delete_from_list.rb:12]: what do you think about moving file_exist(s) into the helpers and removing that duplication?

File: spec/unit/recipes/add_to_list_missing_file_spec.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@
 require 'chefspec'
 require 'chefspec/berkshelf'
 
-describe 'test::add_to_list_missing_file' do
+describe 'test::add_to_list_missing_file_rspec' do
```
Comment (damacus) [inline spec/unit/recipes/add_to_list_missing_file_spec.rb]: rspec?

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add labels to this Pull Request</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: spec/unit/recipes/add_to_list_missing_file_spec.rb (hunk 3 above)
Comment (MarkGibbons) [reply spec/unit/recipes/add_to_list_missing_file_spec.rb]: Working on the names.  Those files get used by the rspec tests.  I'll combine the other files.

File: resources/delete_from_list.rb (hunk 2 above)
Comment (MarkGibbons) [reply resources/delete_from_list.rb:12]: Sure

File: .kitchen.dokken.yml (hunk 1 above)
Comment (MarkGibbons) [reply .kitchen.dokken.yml]: Cool, that is a lot of duplicate work.

File: spec/unit/recipes/add_to_list_missing_file_spec.rb (hunk 3 above)
Comment (damacus) [reply spec/unit/recipes/add_to_list_missing_file_spec.rb]: In that case shouldn't they be "inspec"? 

Having the name at the end seems superfluous .

Comment (MarkGibbons) [reply spec/unit/recipes/add_to_list_missing_file_spec.rb]: I'm going to move them over to the spec directory. They get used by running rspec.
I'll fix the names.

Comment (MarkGibbons) [reply spec/unit/recipes/add_to_list_missing_file_spec.rb]: fixed

File: .kitchen.dokken.yml (hunk 1 above)
Comment (MarkGibbons) [reply .kitchen.dokken.yml]: Done.

**Review by damacus — APPROVED** [review]

**Pull request merged**